### PR TITLE
Add delete by tag feature

### LINKED ISSUE #105: As a user who has used the app a few times now, I am familiar with how to create categories and I want to be able to delete the categories
`delete` command being able to delete all `flashcards` by a specified `tag`

### PR DESCRIPTION
**Pull request by joshtyf**
Implemented delete by tag functionality
DG and UG updates have not been committed yet

Update: DG and UG updates have been committed

### CODE REVIEW

File: src/main/java/quickcache/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -18,36 +22,98 @@
...
+
+    public static DeleteCommand withPredicate(FlashcardPredicate predicate, Set<Tag> tagsToMatch) {
+        return new DeleteCommand(predicate, tagsToMatch);
+    }
+
+    private String createDeleteWithTagsMessage() {
+        StringBuilder sb = new StringBuilder("with tags ");
+        assert isDeleteByTag && tagsToMatch != null;
+        tagsToMatch.forEach(tag -> sb.append(tag.toString()).append(" "));
+        return sb.toString().trim();
```
Comment (josiahkhoo) [inline src/main/java/quickcache/logic/commands/DeleteCommand.java]: Perhaps you can make use of `.reduce()` to make this method even more concise.
```java
return tagsToMatch.stream().reduce(("with tags ", (curr, next) -> curr + next.toString() + " ").trim()
```

**Review by josiahkhoo — APPROVED** [review]
LGTM, just needs to clear the checkstyle as well on top of the docs addition.

File: src/main/java/quickcache/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (joshtyf) [reply src/main/java/quickcache/logic/commands/DeleteCommand.java]: truly a CS2030 TA, thanks for the suggestion!

**Comment by joshtyf** [comment]
DG and UG updates have been committed

File: src/main/java/quickcache/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -64,10 +64,12 @@ public static DeleteCommand withPredicate(FlashcardPredicate predicate, Set<Tag>
...
     private String createDeleteWithTagsMessage() {
-        StringBuilder sb = new StringBuilder("with tags ");
         assert isDeleteByTag && tagsToMatch != null;
-        tagsToMatch.forEach(tag -> sb.append(tag.toString()).append(" "));
-        return sb.toString().trim();
+        return tagsToMatch.stream()
+                .reduce("with tags ",
+                        (curr, next) -> curr + next.toString() + " ",
+                        String::concat)
+                .trim();
```
Comment (ChenXJ98) [inline src/main/java/quickcache/logic/commands/DeleteCommand.java]: Great implementation!

**Review by ChenXJ98 — APPROVED** [review]
LGTM! 👍

**Pull request merged**